tasks/queue-granules/src/index.ts

```diff
@@ -0,0 +1,235 @@
+  return granuleBatch.map((granule) => ({
+    ...granule,
+    createdAt: granule.createdAt ?? createdAt,
+  }));
+}
+
+/**
+ * See schemas/input.json and schemas/config.json for detailed event description
+ *
+ * @param event - Lambda event object
+ * @returns see schemas/output.json for detailed output schema
+ *   that is passed to the next task in the workflow
+ */
+async function queueGranules(event: HandlerEvent): Promise<QueueGranulesOutput> {
+  const granules = (event.input.granules || []);
+  const memoizedFetchProvider = memoize(fetchGranuleProvider, (_, providerId) => providerId);
+  const memoizedFetchCollection = memoize(
+    collectionsApi.getCollection,
+    ({ collectionName, collectionVersion }) => constructCollectionId(
+      collectionName,
+      collectionVersion
+    )
+  );
+  const parentExecutionArn = buildExecutionArn(
+    get(event, 'cumulus_config.state_machine')!,
+    get(event, 'cumulus_config.execution_name')!
+  )!;
+  const pMapConcurrency = get(event, 'config.concurrency', 3);
+
+  const messageTemplate = await getJsonS3Object(
+    event.config.internalBucket,
+    templateKey(event.config.stackName)
+  );
+  const { arn: granuleIngestWorkflowArn } = await getJsonS3Object(
+    event.config.internalBucket,
+    getWorkflowFileKey(event.config.stackName, event.config.granuleIngestWorkflow)
+  );
+
+  const executionArns = await pMap(
+    createIterable(granules, event.config.preferredQueueBatchSize),
+    async ({ provider, collectionId, chunks }) => {
+      const { name: collectionName, version: collectionVersion } = deconstructCollectionId(
+        collectionId
+      );
+      const [collection, normalizedProvider] = await Promise.all([
+        memoizedFetchCollection({
+          prefix: event.config.stackName,
+          collectionName,
+          collectionVersion,
+        }),
+        memoizedFetchProvider(event, provider),
+      ]);
+
+      return await pMap(
+        chunks,
+        async (granuleBatchIn) => {
+          const granuleBatch = updateGranuleBatchCreatedAt(granuleBatchIn, Date.now());
+          await pMap(
+            granuleBatch,
+            (queuedGranule) => {
+              const { granuleId, updatedAt, createdAt } = queuedGranule;
+
+              if (updatedAt && (!Number.isInteger(updatedAt) || updatedAt < 0)) {
+                throw new Error(`Invalid updatedAt value: ${queuedGranule.updatedAt} `
+                                + `for granule with granuleId: ${queuedGranule.granuleId}`);
+              }
+              return granulesApi.updateGranule({
+                prefix: event.config.stackName,
+                collectionId,
+                granuleId,
+                body: {
+                  collectionId,
+                  granuleId,
+                  status: 'queued',
+                  // We use the non-null assertion operator below because we know that
```

Comment:
I ended up removing the comment (https://github.com/nasa/cumulus/pull/3441/commits/ca3b2a8c4120e0c5d1eccae89e10bf67af54495c) because of this change:
https://github.com/nasa/cumulus/pull/3441/commits/3b2e55ce3c3210dc8130a7ed9ec9e5a82c40790c#diff-5a524644461453261cb8eed4a96282c50fcd293225634ffdd3ea700eae8d13a8R181